### ISSUE TITLE
[Bug]: Broadcasting one seeded SamplingParams across a batch gives every prompt the same noise vector

### ISSUE THREAD

**Issue by tolleybot** [vllm-project/vllm#50440]
### Summary

When a batch of prompts is submitted to the offline API with one seeded `SamplingParams` (`LLM.generate(prompts, sampling_params)` / `LLM.chat`), vLLM broadcasts the same object — and therefore the same seed — to every request in the batch (`OfflineInferenceMixin._params_to_seq` returns `[params] * num_requests`). All trajectories then draw an identical noise vector: their sampled tokens are correlated, not independent draws from `softmax(logits)`. Over multiple trajectories, this biases the marginal token distribution rather than averaging out.

Scope notes:

- `n>1` within a request is already handled: `ParentRequest._get_child_sampling_params` derives `seed + index` per child.
- Passing an explicit `Sequence[SamplingParams]` whose seeds already differ avoids the correlation, but the natural broadcast path silently produces it. An explicit list that repeats the same seed hits the bug just the same.
- The two call forms are also inconsistent: `generate([a], SamplingParams(seed=42))` and `generate([a], [SamplingParams(seed=42)])` are the same request expressed two ways, and any per-request seed derivation has to apply to both or they will diverge.
- A fix should derive each request's seed from `(seed, prompt)` rather than from batch position, so a prompt's result does not depend on batch composition or order.

### Affected code

`vllm/entrypoints/offline_utils.py` — `_params_to_seq` broadcasts one `SamplingParams` object (one seed) to all requests:

```python
return [params] * num_requests
```

`vllm/v1/sample/ops/topk_topp_sampler.py` — where the shared seed becomes a shared noise vector:

- `random_sample()` — per-request generator loop:

```python
q[i].exponential_(generator=generator)   # one Exp(1) per (request, token_id)
```

- `sample_with_exponential_noise()` — the argmax:

```python
probs.div_(q).argmax(dim=-1).view(-1)
```

### Mechanism

Random sampling uses the Gumbel-max trick in the probability domain: draw one `Exp(1)` variable `q` per token, then take `argmax(probs / q)` — algebraically `argmax(log p + g)` with `g = -log(q) ~ Gumbel(0,1)`.

Each request initializes its own `torch.Generator` from the integer `seed` **once at request entry**, reused across decode steps (prefill consumes no RNG). Consequently, at step 1:

- Every request with `seed=N` has the same generator state.
- Each request draws a vocab-sized exponential vector `q` that is identical across all same-seed requests.
- The argmax depends on `log p - log q`. The signal `log p` is per-prompt; the noise `q` is shared across all same-seed requests.

### Expected vs actual

- **Expected:** broadcasting a seeded `SamplingParams` gives each prompt an independent noise stream, derived deterministically from `(seed, prompt)` so results are reproducible and independent of batch composition.
- **Actual:** every prompt in the batch receives the identical seed and therefore the identical noise vector.

**Comment by Mahnoor-Zaffar** [comment]
I'm excited to tackle this issue! The problem arises from using a shared seed across trajectories, leading to correlated sampling. I plan to refactor the sampling logic to create a unique random generator for each trajectory by adjusting the seed. This will ensure independent sampling and eliminate structural bias. Please assign me this task!

**Comment by tolleybot** [comment]
@Mahnoor-Zaffar 
> I'm excited to tackle this issue! The problem arises from using a shared seed across trajectories, leading to correlated sampling. I plan to refactor the sampling logic to create a unique random generator for each trajectory by adjusting the seed. This will ensure independent sampling and eliminate structural bias. Please assign me this task!

thanks for the interest.  I already started a PR but feel free to review and let me know your thoughts.
